lib/pure/parsesql.nim

```diff
@@ -645,8 +647,27 @@ proc parseDataType(p: var SqlParser): SqlNode =
       eat(p, tkParRi)
   else:
     expectIdent(p)
-    result = newNode(nkIdent, p.tok.literal)
+    let isNumericDataType = isKeyw(p, "fixed") or 
```

Comment:
if there's any chance this could be a hotopt, there's a more efficient way to do this, if not then that's fine